lib/db/memory.js

```diff
@@ -265,36 +265,42 @@ MemoryStore.prototype = {
     for (var i = 0; i < ids.length; i++) {
       var id = ids[i];
       if (this.tokens[id].userId.toString('hex') === uid) {
-        activeClientIds.push(this.tokens[id].clientId);
+        activeClientIds.push({clientid: this.tokens[id].clientId, clientTokenTime: this.tokens[id].createdAt});
       }
     }
 
     // unique clients
-    activeClientIds.forEach(function (clientIdBuf) {
-      var clientIdHex = unbuf(clientIdBuf);
-      if (! findByBufProp(activeClients, 'id', clientIdHex)) {
-        var c = self.clients[clientIdHex];
+    activeClientIds.forEach(function (clientBuf) {
+      var clientIdHex = unbuf(clientBuf.clientid);
+      var clientTokenTime = unbuf(clientBuf.clientTokenTime);
+      var index = findByBufProp(activeClients, 'id', clientIdHex);
```

Comment:
After that you can change `var activeClients = [];` to an object and look up that object by clientId. Instead of using `.push` you would use `activeClients[clientId] = client`